js/main.js

```diff
@@ -0,0 +1,84 @@
+// Массив с описанием фото
+const DESCRIPTION = ['Описание 1','Описание 2','Описание 3','Описание 4','Описание 5','Описание 6','Описание 7','Описание 8',
+'Описание 9','Описание 10','Описание 11','Описание 12','Описание 13','Описание 14','Описание 15','Описание 16','Описание 17',
+'Описание 18','Описание 19','Описание 20','Описание 21','Описание 22','Описание 23','Описание 24','Описание 25'];
+
+// Массив с именами комментаторов
+const NAMES_FOR_COMMENTS = ['Морти', 'Рик', 'Влада', 'Ника', 'Арагорн', 'Фродо'];
+
+// Массив с текстами комментариев
+const TEXT_FOR_COMMENTS = [
+  'Всё отлично!',
+  'В целом всё неплохо. Но не всё.',
+  'Когда вы делаете фотографию, хорошо бы убирать палец из кадра. В конце концов это просто непрофессионально.',
+  'Моя бабушка случайно чихнула с фотоаппаратом в руках и у неё получилась фотография лучше.',
+  'Я поскользнулся на банановой кожуре и уронил фотоаппарат на кота и у меня получилась фотография лучше.',
+  'Лица у людей на фотке перекошены, как будто их избивают. Как можно было поймать такой неудачный момент?!'
+];
+
+// Генератор ID для комментариев с уникальным значением
+const createRandomIdGenerator = (min, max) => {
+  const previousValues = [];
+
+  return function () {
+    let currentValue = getRandomNum(min, max);
+    if (previousValues.length >= (max - min + 1)) {
+      return null;
+    }
+    while (previousValues.includes(currentValue)) {
+      currentValue = getRandomNum(min, max);
+    }
+    previousValues.push(currentValue);
+    return currentValue;
+  };
+}
+
+const createComment = () => ({
+    id: getRandomNum(1, 1000),
+    avatar: getRandomNum(1, 6), // 'img/avatar-' + getRandomNum(1, 6) + '.svg' по заданию не понял какой вариант верный
+    message: TEXT_FOR_COMMENTS[getRandomNum(0, TEXT_FOR_COMMENTS.length -1)],
+    name: NAMES_FOR_COMMENTS[getRandomNum(0, NAMES_FOR_COMMENTS.length -1)]
+  });
+
+// генератор случайного кол-ва комментариев под каждым фото
+const commentsGenerator= () => {
+  const COMMENTS_FOR_PHOTO = [];
+    let i = 0;
+    let numberOfComments = getRandomNum(1, 4);
+    while (i < numberOfComments) {
+      COMMENTS_FOR_PHOTO.push(createComment());
+        i++
+  }
+        return COMMENTS_FOR_PHOTO;
+}
```

Comment:
В целом `while` -- это нормальный вариант, но он не такой декларативный, читабельный и вообще циклы `for/while` И подобные используются в основном в алгоритмах, там где они действительно необходмиы, в каких-то универсальных функциях и там где нужна производительность на доли секунд. В общем, с массивами лучше использовать `forEach` метод масива. Функция, как и весь код, должна читаться как текст, например
```suggestion
// Константы в самый верх файла
const MIN_COMMENTS_COUNT = 1;
const MAX_COMMENTS_COUNT = 4;

const createComments = () => {
  return Array.from({length: getRandomNumber(MIN_COMMENTS_COUNT, MAX_COMMENTS_COUNT)}, createComment);
};
```
Т.е. читается как `функция создать комментарии, возвращает массив случайной длины от MIN_COMMENTS_COUNT до MAX_COMMENTS_COUNT, а каждый элемент массива -- комментарий созданный функцией createComment`

Reply:
В целом, если посмотреть теперь на эту функцию в плане дальнейшего развития, то можно вынести из неё длину получаемого массива комментариев в аргументы, чтобы вовсе сделать функцию универсальной
```suggestion
const createComments = (length) => {
  return Array.from({length}, createComment);
};
```

И передавать уже при вызове где нужно количество сколько нужно в конкретный моммент вызова мне комментариев, например:
```ts
const comments = createComments(5);
//или 
const comments = createComments(500);
// или наш случай
const comments = createComments(getRandomNumber(MIN_COMMENTS_COUNT, MAX_COMMENTS_COUNT));
```